### PR TITLE
Modernize static Proxy home page

### PR DESCRIPTION
**Pull request by ncounter**
## Reviewers

- @mcalmer please take a look only at the overview change if it makes sense or not
- @juliogonzalez please take a look at the `.spec` only, that's something I think you have the best up-to-date knowledge
- @cbbayburt pretty trivial and straight forward static code, but it deserves a review anyway

Thank you

## What does this PR change?

**Context**: when accessing a proxy system via `http://<proxy_ip_address>` the machine provides a static homepage where there is a link to the filesystem `/pub` directory content. The page was really outdated (showing a SUSE Manager 1.2 logo page).

This PR is about to wildly refactoring of the static proxy homepage. In addition to that, there are changes to the `.spec` file in order to serve a *suse_proxy* or a *uyuni_proxy* branded homepage depending on the product being installed.

**Drawback**: being the page a complete static resource and not willing to invest more in this to make it dynamic anyway for now, the code has to be duplicated and customized for both the product versions. But that's really a small chunk of code. Whenever it will grow, a proper dynamic switching and code-sharing should be implemented instead.

## GUI diff

Before:
Screenshot from 2021-02-08 11-20-14

After:
Screenshot from 2021-02-08 11-20-25

or
Screenshot from 2021-02-08 15-09-41

- [x] **DONE**

## Documentation
- No documentation needed: @mcalmer @Loquacity do we have this documented anywhere?

- [x] **DONE**

## Test coverage
- No tests: static files not tested

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/12686
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: proxy/proxy-html/spacewalk-proxy-html.spec [hunk 1]
```diff
@@ -49,18 +56,17 @@ displays, if you navigate to it using your browser.
...
-%{htmldir}/_rhn_proxy/*.ico
-%{htmldir}/_rhn_proxy/*.png
+%dir %{htmldir}
+%config %{htmldir}/index.html
+%{htmldir}/sources/img/*.ico
+%{htmldir}/sources/img/*.png
 %doc LICENSE
 %if 0%{?suse_version}
-%dir %dir %{htmldir}/_rhn_proxy
+%dir %dir %{htmldir}
```
Comment (juliogonzalez) [inline proxy/proxy-html/spacewalk-proxy-html.spec:69]: You are already included `%{htmldir} at line 64 for all OS.

Looks to me that this whole if can go away?

**Review by juliogonzalez — COMMENTED** [review]
One comment for the spec.

**Comment by Loquacity** [comment]
> * No documentation needed: @mcalmer @Loquacity do we have this documented anywhere?

I don't think so. Perhaps @keichwa can think of something I'm missing though.

**Comment by keichwa** [comment]
thanks for pinging.  No, we do not mention the (old) proxy home page in the docs.  I'm not sure whether the user has to know about it.

File: proxy/proxy-html/spacewalk-proxy-html.spec (hunk 1 above)
Comment (ncounter) [reply proxy/proxy-html/spacewalk-proxy-html.spec:69]: Good point @juliogonzalez , it was the case when there was the subfolder `_rhn_proxy` before this PR, but now it is removed and the whole `if` condition can be dropped. Thanks

File: proxy/proxy/httpd-conf/spacewalk-proxy-wsgi.conf [hunk 2]
```diff
@@ -21,7 +21,7 @@ WSGIPythonPath "/usr/share/rhn"
 </IfVersion>
 
 <LocationMatch "^/*">
-    DirectoryIndex _rhn_proxy/index.html index.html index.htm index.html.var index.shtml index.php index.php4 index.php3 index.phtml index.cgi
+    DirectoryIndex suse_proxy/index.html index.html index.htm index.html.var index.shtml index.php index.php4 index.php3 index.phtml index.cgi
```
Comment (mcalmer) [inline proxy/proxy/httpd-conf/spacewalk-proxy-wsgi.conf:24]: I think this is wrong. It seems that `suse_proxy` and `uyuni_proxy` only exists on the sources.
After installation, its exists directly in the root dir, right?

I think `suse_proxy/index.html` can be removed

File: proxy/proxy/httpd-conf/spacewalk-proxy.conf [hunk 3]
```diff
@@ -39,7 +39,7 @@
     SetHandler None
 </LocationMatch>
 
-<LocationMatch "^/_rhn_proxy">
+<LocationMatch "^/suse_proxy">
```
Comment (mcalmer) [inline proxy/proxy/httpd-conf/spacewalk-proxy.conf:42]: I think this can be removed as well. I think there is no `suse_proxy` in the installed system.

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: proxy/proxy/httpd-conf/spacewalk-proxy-wsgi.conf (hunk 2 above)
Comment (ncounter) [reply proxy/proxy/httpd-conf/spacewalk-proxy-wsgi.conf:24]: Good catch, thanks for pointing it out. Initially I only renamed it, but then I went for the double option removing the subfolder, so yes everything goes in the web root (which is `/srv/www/htdocs`). Fixed now :+1:

File: proxy/proxy/httpd-conf/spacewalk-proxy.conf (hunk 3 above)
Comment (ncounter) [reply proxy/proxy/httpd-conf/spacewalk-proxy.conf:42]: Fixed :+1:

**Review by cbbayburt — APPROVED** [review]
LGTM :+1: (I reviewed the frontend only)

BTW, that favicon is also outdated as hell :rofl: 
It's not relevant for this PR though, it's the same in the server anyway.

**Review by mcalmer — APPROVED** [review]

**Comment by ncounter** [comment]
> BTW, that favicon is also outdated as hell rofl
> It's not relevant for this PR though, it's the same in the server anyway.

I updated it anyway. Time to touch this path once and never go back again :wink:

**Comment by juliogonzalez** [comment]
> I updated it anyway. Time to touch this path once and never go back again wink

By "never" you mean until next year, right? :trollface: 

Remember to add the `merge-candidate` label if you want this merged soon.

**Pull request merged**